### ISSUE TITLE
diamond view - various bugs

### ISSUE THREAD

**Issue by Ferrisx4** [bbuchfink/diamond#279]
When attempting to use `diamond view...` on .daa files generated by diamond, we experience various issues including 
- segmentation faults
- core dumps
- unexpected end of file
- a combination of these

These issues seem to occur regardless of data being used (both query and target) and regardless of OS—Centos (AMD CPUs) and MacOS (Intel CPUs). Upon repeated runs using the same .daa file, different errors happen.

I'm attempting to create, from the diamond archive file (which gets generated always without issue), a BLAST pairwise format:
`--outfmt 0`
and a BLAST tabular-like format:
`--outfmt 6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore sseq qseq`
separately in two different calls to `diamond view`

We're currently running the latest version 0.9.24.125 (cloned and compiled yesterday, April 22 2019) although this was happening previously (but still with 0.9.24).

Example of script being used:
```
diamond blastp \
    --threads 16 \
    --db Poep.1_0.peptides.dmnd \
    --query query_33.fasta \
    --more-sensitive \
    -k 5 \
    -e 0.003 \
    --subject-cover 50 \
    --query-cover 50 \
    --outfmt 100  \
    --out results.daa 

diamond view \
    --daa results.daa  \
    --outfmt 6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore sseq qseq \
    --out results_blast.txt

diamond view \
    --daa results.daa  \
    --outfmt 0  \
    --out results_pairwise.txt
```

**Comment by bbuchfink** [comment]
Thanks for pointing this out, there seems to be a serious bug in there. I will work on fixing it. The bug does not appear to be present in the 0.9.24 release version, so please use this in the meantime instead of cloning the repo.

**Comment by Ferrisx4** [comment]
Thanks, the release version does indeed work without issue.

I was slightly surprised, however, that Diamond complained that the index was generated with a different version of Diamond. 
Using the conversion method in https://github.com/bbuchfink/diamond/issues/227#issuecomment-476751086 worked so no worries.

**Comment by bbuchfink** [comment]
This is fixed in ac3051fea14598ec2e8a834983210be74aecf8fd.
